b2/src/main/java/org/jclouds/b2/features/ObjectApi.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.b2.features;
+
+import static javax.ws.rs.core.MediaType.APPLICATION_JSON;
+
+import static org.jclouds.blobstore.attr.BlobScopes.CONTAINER;
+
+import javax.inject.Named;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+
+import org.jclouds.Fallbacks.NullOnNotFoundOr404;
+import org.jclouds.blobstore.attr.BlobScope;
+import org.jclouds.http.options.GetOptions;
+import org.jclouds.javax.annotation.Nullable;
+import org.jclouds.b2.binders.UploadFileBinder;
+import org.jclouds.b2.domain.B2Object;
+import org.jclouds.b2.domain.B2ObjectList;
+import org.jclouds.b2.domain.DeleteFileResponse;
+import org.jclouds.b2.domain.HideFileResponse;
+import org.jclouds.b2.domain.UploadFileResponse;
+import org.jclouds.b2.domain.UploadUrlResponse;
+import org.jclouds.b2.filters.RequestAuthorization;
+import org.jclouds.b2.filters.RequestAuthorizationDownload;
+import org.jclouds.b2.functions.ParseB2ObjectFromResponse;
+import org.jclouds.rest.annotations.Fallback;
+import org.jclouds.rest.annotations.MapBinder;
+import org.jclouds.rest.annotations.PayloadParam;
+import org.jclouds.rest.annotations.RequestFilters;
+import org.jclouds.rest.annotations.ResponseParser;
+import org.jclouds.rest.binders.BindToJsonPayload;
+
+@BlobScope(CONTAINER)
+public interface ObjectApi {
+   @Named("b2_get_upload_url")
+   @POST
+   @Path("/b2api/v1/b2_get_upload_url")
+   @RequestFilters(RequestAuthorization.class)
+   @MapBinder(BindToJsonPayload.class)
+   @Consumes(APPLICATION_JSON)
+   @Produces(APPLICATION_JSON)
+   UploadUrlResponse getUploadUrl(@PayloadParam("bucketId") String bucketId);
```

Comment:
I'm not sure about this -- users will likely directly feed the result of `getUploadUrl` into `uploadFile`.  Throwing `ContainerNotFoundException` in the former gives more information that the `NullPointerException` from the latter.